syntaxes/happy.YAML-tmLanguage

```diff
@@ -0,0 +1,75 @@
+fileTypes:
+      - y
+name: Haskell Happy
+patterns:
+  - include: "#comments"
+  - include: "#blocks"
+  - include: "#syntax"
+  - include: "#strings"
+repository:
+  comments:
+    patterns:
+      - name: comment.line
+        match: \-\-.*$
+      - name: comment.block
+        begin: "{-"
+        end: "-}"
+  blocks:
+    name: meta.embedded.block.haskell
+    begin: "{"
+    beginCaptures:
+      "0": { name: punctuation.block.begin.happy }
+    end: "}"
+    endCaptures:
+      "0": { name: punctuation.block.end.happy }
+    patterns:
+      - include: source.haskell
+  syntax:
+    patterns:
+      - name: keyword.operator.pragma.happy
+        match: \%\w+
+      - name: variable.parameter.happy
+        match: \$\w+
```

Comment:
Should this be part of the Haskell blocks instead? I assume it 's meant to catch parameters like `$1`, `$2` etc, but it doesn't fire because those happen inside the Haskell blocks.